backend/internal/auth/sessionmanager.go

```diff
@@ -44,3 +53,39 @@ func (sm *SessionManager) CreateSession(account model.Account) (Session, error)
 
 	return s, nil
 }
+
+// GetSession gets a session by sessionID if it exists and isn't expired, otherwise
+// it returns an empty Session object and a non-nil error
+func (sm *SessionManager) GetSession(sid SessionID) (Session, error) {
+	sm.mtx.RLock()
+	session, ok := sm.store[sid]
+	sm.mtx.RUnlock()
+
+	if !ok {
+		return Session{}, ErrSessionDNE
+	}
+
+	if time.Now().After(session.Expires) {
+		return Session{}, ErrSessionTimeout
+	}
+
+	return session, nil
+}
+
+// DeleteSession deletes a session from the session manager. Returns true if the session
+// was found and deleted, or false if the session wasn't found
+func (sm *SessionManager) DeleteSession(sid SessionID) bool {
+	// Check whether the session exists and return false if it doesn't
+	sm.mtx.RLock()
+	_, ok := sm.store[sid]
+	sm.mtx.RUnlock()
+	if !ok {
+		return ok
+	}
+	// Session does exist, delete it
+	sm.mtx.Lock()
+	defer sm.mtx.Unlock()
+	delete(sm.store, sid)
```

Comment:
nit: all of this should happen under a single write lock
if you have 2 concurrent `DeleteSession` calls, they might intertwine their read/write locks and return the wrong bool.

Reply:
Good catch